main.tf

```
@@ -11,7 +11,10 @@ locals {
   ami                    = var.ami != "" ? var.ami : join("", data.aws_ami.default.*.image_id)
   ami_owner              = var.ami != "" ? var.ami_owner : join("", data.aws_ami.default.*.owner_id)
   root_volume_type       = var.root_volume_type != "" ? var.root_volume_type : data.aws_ami.info.root_device_type
-  public_dns             = var.associate_public_ip_address && var.assign_eip_address && module.this.enabled ? data.null_data_source.eip.outputs["public_dns"] : join("", aws_instance.default.*.public_dns)
+  public_dns = (var.associate_public_ip_address && var.assign_eip_address && module.this.enabled ?
+    ("ec2-${replace(join("", aws_eip.default.*.public_ip), ".", "-")}.${local.region == "us-east-1" ?
+    "compute-1" : "${local.region}.compute"}.amazonaws.com") :
+  join("", aws_instance.default.*.public_dns))
```

While I appreciate the effort, this is really hard to read. Let's use local variables to simplify the expression.
```suggestion

  region_domain  = local.region == "us-east-1" ? "compute-1.amazonaws.com" : "${local.region}.compute.amazonaws.com"
  eip_public_dns = "ec2-${replace(join("", aws_eip.default.*.public_ip), ".", "-")}.${local.region_domain}"
  public_dns     = (
    var.associate_public_ip_address && var.assign_eip_address && module.this.enabled ?
    local.eip_public_dns : join("", aws_instance.default.*.public_dns)
  )
```